6.3..js

```diff
@@ -0,0 +1,48 @@
+function logFive(sequence) {
+    for (var i = 0; i < 5; i++) {
+        if (!sequence.next()) {
+            break;
+        } else {
+            console.log(sequence.current());
+        }
+    }
+}
+
+function ArraySeq(array) {
+    this.pos = -1;
+    this.array = array;
+}
+
+
+ArraySeq.prototype.next = function() {

```

Comment:
nie widze tego wykorzystania. jak pokazesz to moze bede mogl pomoc. bo nie widze powodow, dla ktorych mialoby nie dzialac.